chain-evm/src/machine.rs

```diff
@@ -70,6 +70,21 @@ pub type GasPrice = U256;
 /// Gas limit for EVM operations.
 pub type GasLimit = U256;
 
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+/// EVM Environment parameters needed for execution.
+pub struct Environment {
+    pub gas_price: GasPrice,
+    pub origin: Origin,
+    pub chain_id: ChainId,
+    pub block_hashes: BlockHashes,
+    pub block_number: BlockNumber,
```

Comment:
This should be taken from the ledger state and updated every time there is a new block.

Reply:
Fine, then go for it :)